Docs:  Update INSTALL.md to add link to build notes

ABISprotocol opened the pull request.

ABISprotocol committed:
Update INSTALL.md to add link to developer notes

MarcoFalke commented on INSTALL.md:
```diff
@@ -1,5 +1,5 @@
 ================
 
-See doc/build-*.md for instructions on building the various
+See doc/build-*.md or [link here](https://github.com/bitcoin/bitcoin/blob/57b34599b2deb179ff1bd97ffeab91ec9f904d85/doc/README.md#building) for instructions on building the various
```
Please make the link relative to the git root:

```md
[link](/doc/README.md)

laanwj replied:
Right, including a specific commit is id not a good idea because it essentially freezes the build instructions in time.

ABISprotocol replied:
Thanks, am correcting.

ABISprotocol committed:
Minor changes to INSTALL.md

This changes patch-1 branch as originally proposed and suggests minor changes to INSTALL.md

laanwj commented:
Now you removed the link completely, why?

ABISprotocol commented:
@laanwj I was under the impression from the discussion the link wasn't preferred but if it is considered to be alright then I can walk it back in.

laanwj commented:
>  I can change it to link relative to git root as @MarcoFalke suggested.

Yes, why not just follow his suggestion. Our only comment was that you shouldn't link to a specific revision, nor have to include the github part.

laanwj commented:
Going to close this. It's taking too long for a one-line change.

ABISprotocol commented:
Sorry about the length of time @laanwj I got insanely busy and then literally couldn't get back to it.  My sincere apologies.  Some other time perhaps on something similar.  Cheers and happy 4th.
